include/tc/core/cuda/cuda_tc_executor.h

```diff
@@ -23,20 +28,18 @@
 #include "tc/core/utils/dlpack.h"
 #include "tc/lang/parser.h"
 
-#include <dlpack/dlpack.h>
-
 namespace tc {
 
 class CudaTcExecutor : public ::tc::TcExecutor {
  public:
   CudaTcExecutor(
-      const std::string& def,
-      const std::vector<const DLTensor*>& inputsInfo)
-      : TcExecutor(def, inputsInfo) {}
-  CudaTcExecutor(
-      lang::TreeRef tree,
-      const std::vector<const DLTensor*>& inputsInfo)
-      : TcExecutor(tree, inputsInfo) {}
+      std::string id,
+      std::vector<const DLTensor*> inputsInfo,
+      const std::string& options,
+      lang::TreeRef tcDefinition,
```

Comment:
`const`?

Reply:
```lang::TreeRef``` is used everywhere with by-value semantics, keeping the same behavior.
Underlying is a ```shared_ptr``` IIRC.

Reply:
yup, it's a `shared_ptr`